File: src/main/java/fi/metatavu/metaform/server/settings/SystemSettingController.java
package fi.metatavu.metaform.server.settings;

import org.eclipse.microprofile.config.ConfigProvider;

import javax.enterprise.context.ApplicationScoped;

/**
 * Controller for system settings.
 * 
 * @author Antti Leppä
 */
@ApplicationScoped
public class SystemSettingController {
  
  /**
   * Returns whether the system is running in test mode or not
   * 
   * @return whether the system is running in test mode or not
   */
  public boolean inTestMode() {
    return "TEST".equals(ConfigProvider.getConfig().getValue("runmode", String.class));
  }
  
}

File: src/main/kotlin/fi/metatavu/metaform/server/controllers/SystemSettingController.kt
package fi.metatavu.metaform.server.controllers

import org.eclipse.microprofile.config.ConfigProvider
import javax.enterprise.context.ApplicationScoped

/**
 * Controller for System settings
 */
@ApplicationScoped
class SystemSettingController {

    /**
     * Returns whether the system is running in test mode or not
     *
     * @return whether the system is running in test mode or not
     */
    fun inTestMode(): Boolean {
        return "TEST" == ConfigProvider.getConfig().getValue("runmode", String::class.java)
    }
}